File: apps/agent-prototype-baml/app/schemas/event.py
from datetime import datetime
from typing import Optional
from app.schemas.base import BaseSchema

class EventBase(BaseSchema):
    user_id: Optional[str] = None
    title: str
    description: Optional[str] = None
    start_time: datetime
    end_time: datetime
    location: Optional[str] = None

class EventCreate(EventBase):
    id: Optional[str] = None

class Event(EventBase):
    id: str
    created_at: datetime

File: apps/frontend/lib/core/common/domain/models/event.dart
import 'package:flutter/material.dart';

class Event {
  final String id;
  final String userId;
  final String title;
  final String? description;
  final DateTime startTime;
  final DateTime endTime;
  final String? location;
  final Color? color;
  final bool isAllDay;
  final DateTime createdAt;
  final DateTime updatedAt;

  Event({
    required this.id,
    required this.userId,
    required this.title,
    this.description,
    required this.startTime,
    required this.endTime,
    this.location,
    this.color,
    this.isAllDay = false,
    required this.createdAt,
    required this.updatedAt,
  });
}
